src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs

```diff
@@ -285,40 +283,40 @@ private static unsafe bool TryParseNumber(ref char* str, char* strEnd, NumberSty
             }
 
             int state = 0;
-            char* p = str;
-            char ch = p < strEnd ? *p : '\0';
-            char* next;
+            ReadOnlySpan<char> p = str;
+            char ch = p.IsEmpty ? '\0' : p[0];
+            int? nextIndex;
```

Comment:
I wonder if making this nullable will have an effect on perf? We'll see in the benchmarks I guess